grpc/src/main/java/com/linecorp/armeria/internal/common/grpc/GrpcCancellingExceptionPredicateProvider.java

```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright 2023 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package com.linecorp.armeria.internal.common.grpc;
+
+import java.util.function.Predicate;
+
+import com.linecorp.armeria.internal.common.util.CancellingExceptionPredicateProvider;
+
+import io.grpc.Status.Code;
+import io.grpc.StatusException;
+import io.grpc.StatusRuntimeException;
+
+public final class GrpcCancellingExceptionPredicateProvider implements CancellingExceptionPredicateProvider {
+    @Override
+    public Predicate<Throwable> newPredicate() {
+        return cause -> {
+            if (cause instanceof StatusRuntimeException) {
```

Comment:
Could we also take `ArmeriaStatusException` into account?

Reply:
That's a good idea. 👍